### PR TITLE
cutter: 1.12.0 -> 2.0.0 

### PR DESCRIPTION
**Pull request by Mic92**
###### Motivation for this change

@offlinehacker I propose renaming cutter to tcp-cutter. It is less popular than this https://github.com/rizinorg/cutter,
which was originally named radare2-cutter. Since they no longer part of the radare2 project and now named their own
fork rizin, this no longer holds true.
Having it named `tcp-cutter` also has the advantage that the package name matches the binary so that `nix run` just works.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/analysis/rizin/cutter.nix [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+, wrapQtAppsHook
+, cmake
+}:
+
+mkDerivation rec {
+  pname = "cutter";
+  version = "2.0.0";
+
+  src = fetchFromGitHub {
+    owner = "radareorg";
```
Comment (wargio) [inline pkgs/development/tools/analysis/rizin/cutter.nix:18]: `rizinorg` here ?

File: pkgs/development/tools/analysis/rizin/cutter.nix [hunk 2]
```diff
@@ -0,0 +1,47 @@
+{ fetchFromGitHub, lib, mkDerivation
+# nativeBuildInputs
+, qmake, pkg-config
+# Qt
+, qtbase, qtsvg, qtwebengine, qttools
+# buildInputs
+, rizin
+, python3
+, wrapQtAppsHook
+, cmake
```
Comment (tnias) [inline pkgs/development/tools/analysis/rizin/cutter.nix:10]: Those seem to be `nativeBuildInputs` as well.

**Comment by tnias** [comment]
I can build and run it. I tested basic functionality, such as loading a binary, running analysis, scrolling though disassembly/graph/hexdump/string. :+1:

**Comment by tnias** [comment]
Cutter seem to have [translations for many languages](https://github.com/rizinorg/cutter-translations), but `Edit > Preferences > Appearance > Language` only offers English.

**Comment by tnias** [comment]
I packaged a plugin rizin/cutter plugin [rz-ghidra.nix](https://gist.github.com/tnias/dee7e9b1e7af882cc427f00709fbea76). Manually copying the artifacts into one of the search paths works. Search paths are fetched in [getPluginDirectories](https://github.com/rizinorg/cutter/blob/18124723c99bd4fdc1d50e894da864162b1cf6d1/src/plugins/PluginManager.cpp#L95) and depend on environment variables that are set in cutters qt application wrapper.

How would I add the plugin in a elegant way? Adding it as a dependency results in an infinite recursion.

**Comment by Mic92** [comment]
> I packaged a plugin rizin/cutter plugin [rz-ghidra.nix](https://gist.github.com/tnias/dee7e9b1e7af882cc427f00709fbea76). Manually copying the artifacts into one of the search paths works. Search paths are fetched in [getPluginDirectories](https://github.com/rizinorg/cutter/blob/18124723c99bd4fdc1d50e894da864162b1cf6d1/src/plugins/PluginManager.cpp#L95) and depend on environment variables that are set in cutters qt application wrapper.
> 
> How would I add the plugin in a elegant way? Adding it as a dependency results in an infinite recursion.

There could be a wrapper around cutter that sets CUTTER_EXTRA_PLUGIN_DIRS to the plugins you specified.

**Comment by Mic92** [comment]
Regarding translations: https://github.com/rizinorg/cutter/blob/3a32ea741e58284e295b72d4cb38bf829ff0d43b/cmake/Translations.cmake#L31
Maybe qt does not look in the right directories? Not sure how this normally works?

**Comment by Mic92** [comment]
might need XDG_DATA_DIRS: https://github.com/NixOS/nixpkgs/issues/37741#issuecomment-428895286

**Pull request merged**